libs/deal-ticket/src/components/deal-ticket-validation/use-order-margin-validation.tsx

```diff
@@ -0,0 +1,44 @@
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import { AccountType } from '@vegaprotocol/types';
+import { toBigNum } from '@vegaprotocol/react-helpers';
+import type { DealTicketMarketFragment } from '../deal-ticket/__generated___/DealTicket';
+import type { OrderMargin } from '../../hooks/use-order-margin';
+import { usePartyBalanceQuery, useSettlementAccount } from '../../hooks';
+
+interface Props {
+  market: DealTicketMarketFragment;
+  estMargin: OrderMargin | null;
+}
+
+export const useOrderMarginValidation = ({ market, estMargin }: Props) => {
+  const { pubKey } = useVegaWallet();
+
+  const { data: partyBalance } = usePartyBalanceQuery({
+    variables: { partyId: pubKey || '' },
+    skip: !pubKey,
+  });
+
+  const settlementAccount = useSettlementAccount(
+    market.tradableInstrument.instrument.product.settlementAsset.id,
+    partyBalance?.party?.accounts || [],
+    AccountType.ACCOUNT_TYPE_GENERAL
+  );
+  if (settlementAccount) {
```

Comment:
Should we enter this block when we 
settlementAccount.balance is falsy or
settlementAccount.asset.decimals is falsy or
estMargin?.margin is undefined

Reply:
Do you know that you're even right ? At least in case when no settlementAccount, we can assume user should be treated as no having enough asset. I've made smth like that.